go/encode.go

```diff
@@ -23,17 +24,49 @@ func GetByte(buf []byte) byte {
 	return byte(GetUint8(buf))
 }
 
+// SafeGetByte first bounds checks the byte slice to ensure a length of 1
+// then decodes a little-endian byte from the byte slice
+func SafeGetByte(buf []byte) (byte, error) {
+	x, err := SafeGetUint8(buf)
+	if err != nil {
+		return byte(0), fmt.Errorf("Bounds check failed for returning safe byte value from SafeGetUint8")
+	}
+	return byte(x), nil
+}
+
 // GetBool decodes a little-endian bool from a byte slice.
 func GetBool(buf []byte) bool {
 	return buf[0] == 1
 }
 
+// SafeGetBool first bounds checks the byte slice to ensure a length of 1
+// then decodes a little-endian bool from the byte slice
+func SafeGetBool(buf []byte) (bool, error) {
+	if len(buf) != 1 {
```

Comment:
Instead of checking the length exactly, this should probably be `if len(buf) < 1` (same in other functions)

Reply:
Hi Jeroen! Thanks for the review.

> Instead of checking the length exactly, this should probably be `if len(buf) < 1` (same in other functions)

Wouldn't this cause this function to incorrectly pass the bounds check if `buf` is an empty slice?

> How is one supposed to use these functions? Ideally these would be used in safe variants of the generated code.

Yes, you're correct, these would have to be used in the generated code as safe options through a compiler flag, I missed that, I'll work on adding that to the Go code generator.

> Also, there is clearly some functional overlap with #8030 

There is definitely overlap in its utility, but `verifier` is more comprehensive in its implementation in verifying each table, buffer offsets, table depth/complexity, etc. and thus incurs more overhead.

These functions are to safely retrieve scalar values with as minimal overhead as possible, given that the original functions already force bounds checks but panic instead of throw.

However, let me know if you think the project would see further utility in these safe scalar retrieval functions, and I'll work on getting codegen working through an --include-safe flag.

Thank you!